clientlib/src/main/proto/yelp/nrtsearch/luceneserver.proto

```diff
@@ -935,4 +945,14 @@ message ForceMergeDeletesResponse {
         FORCE_MERGE_DELETES_SUBMITTED = 1;
     }
     Status status = 1;
+}
+
+message CustomRequest {
+    string id = 1; // ID defined for custom requests in a plugin
```

Comment:
Just a thought, do we need the `id`? It does make it harder for there to be collisions, but it requires the caller to have more knowledge about the implementation.

Reply:
The caller just needs to additionally know the id along with the path. While it does seem like extra information at the moment I think it is easy to have a situation where two plugins share the same path, e.g. two LTR plugins having an endpoint to add new models. Without this id the plugin implementers would have to consciously add some other identifier to the path itself to make sure that there is no collision and also so that users do not get confused.
This would be applicable only if both plugins are installed on the same cluster, which is also certainly possible if there are experiments being done only on the query side without changing the data.

Reply:
I guess, though the plugin creators would need to consciously select a unique id.

I was just trying to imagine what this would look like if we wanted to create a curl like utility to use this endpoint. Perhaps the id could be interpreted as a path prefix.

Reply:
I added both id and path as url params for the generated grpc-gateway. I was trying to specify params as the body in the request instead of * but that was throwing an error. It would work if params was struct, but the resulting json would be more complicated.